scripts/test

```diff
@@ -51,6 +51,9 @@ compose() {
 (
   cd "${projectRoot}"
 
+  # create test output folder
+  mkdir -p test-output
```

Comment:
Doesn't Sonarcloud use this for its metrics -- where they not running before or..?

Reply:
The way the tests work is that when run they will create the `test-output` folder if it doesn't exist.  But because the tests are running in a docker container, the file is created as `root` user.  This actually prevents the container volume mapping from writing the test outputs to it.

The fix is to just run something like `sudo chmod 777 ./test-output` to change the permissions and re-run the tests again.  You're likely only ever going to see that error the first time you run the tests.

Alternatively that can be avoided by creating the folder in advance of running the containerised tests as they'll be created with an appropriate permission.  So seems a no brainer just to include it in the test script so it's created before the container starts and new devs who clone the repo don't have that pain on first run.

SonarCloud does need that folder and if you look in the Jenkins pipeline code you'll see that before it runs the tests, it does make sure the folder is created first -> https://github.com/DEFRA/ffc-jenkins-pipeline-library/blob/dc769b23423d710593a4ddc176f435f4403afd05/src/uk/gov/defra/ffc/Tests.groovy#L12

So the pipeline already does the equivalent of this.